SDWebImage/SDWebImageCompat.h

```diff
@@ -87,15 +87,11 @@ typedef void(^SDWebImageNoParamsBlock)(void);
 
 FOUNDATION_EXPORT NSString *const SDWebImageErrorDomain;
 
-#ifndef dispatch_queue_async_safe
-#define dispatch_queue_async_safe(queue, block)\
-    if (strcmp(dispatch_queue_get_label(DISPATCH_CURRENT_QUEUE_LABEL), dispatch_queue_get_label(queue)) == 0) {\
+#ifndef dispatch_main_async_safe
+#define dispatch_main_async_safe(block)\
+    if (dispatch_queue_get_label(DISPATCH_CURRENT_QUEUE_LABEL) == dispatch_queue_get_label(dispatch_get_main_queue())) {\
```

Comment:
This should still use strcmp. We just need to remove the `dispatch_queue_async_safe` macro

Reply:
@dreampiggy 🤔 Sorry, I know the compiler something like you said, but in this case, what's the issue have you worried about? If current queue is main queue, `dispatch_queue_get_label(DISPATCH_CURRENT_QUEUE_LABEL)` returns the same pointer with `dispatch_queue_get_label(dispatch_get_main_queue())`.

You mean maybe user create a queue with label the same as main queue?

Reply:
@zhongwuzw We can not assume that libdispatch always use a string literal to create the main queue. There are no any documentation or guard for this.

And actually...Does this cause some performance issue. For short ASCII string using `strcmp`..

Reply:
@dreampiggy First of all, I can push a new commit to use `strcmp`, `strcmp` need to compare 
 at least 10 characters for most of the cases because of `com.apple.*` label name. 

Secondly, I just wonder to know what's the issue you worried about.  The `libdispatch` is open source,please [see](https://github.com/apple/swift-corelibs-libdispatch).
From the `API`, `const char * dispatch_queue_get_label(dispatch_queue_t queue);` return `const char *`, it's immutable, `dispatch_queue_s` struct has a member `dq_label`, see below, 

```
#define _OS_MPSC_QUEUE_FIELDS(ns, __state_field__) \
	struct dispatch_object_s *volatile ns##_items_head; \
	DISPATCH_UNION_LE(uint64_t volatile __state_field__, \
			dispatch_lock __state_field__##_lock, \
			uint32_t __state_field__##_bits \
	) DISPATCH_ATOMIC64_ALIGN; \
	/* LP64 global queue cacheline boundary */ \
	unsigned long ns##_serialnum; \
	const char *ns##_label; \  // Here is the dq_label
	struct dispatch_object_s *volatile ns##_items_tail; \
	dispatch_priority_t ns##_priority; \  
	int volatile ns##_sref_cnt
#endif
```

And please see the source code of `dispatch_queue_create` about how to handle the label.

Above all, I think you would think the worries that you have are unnecessary. 😂 


```
const char *
dispatch_queue_get_label(dispatch_queue_t dq)
{
	if (slowpath(dq == DISPATCH_CURRENT_QUEUE_LABEL)) {
		dq = _dispatch_get_current_queue();
	}
	return dq->dq_label ? dq->dq_label : "";
}
```

Reply:
@zhongwuzw Not worry...Just a little curious about this changes. Because from my personal experience I do not using `==` from C string compare.

I see the source code of libdispatch, it seems that your consideration is adoptable. So we can use this and hope for Apple does not change this implementation. 😅